server.js

```diff
@@ -102,25 +106,43 @@ function setupLogging() {
     setupLogging();
     bundle.bundle("site", outputDir, liveReload);
 
-    analysis.migrateCompression(dataDir, ".json", ".json.br");
-    analysis.migrateCompression(dataDir, ".json.gz", ".json.br");
+    if (!skipDataUpdate) {
+        analysis.migrateCompression(dataDir, ".json", ".json.br");
+        analysis.migrateCompression(dataDir, ".json.gz", ".json.br");
 
-    if (fs.existsSync(`${dataDir}/latest-canonical.json.${analysis.FILE_COMPRESSOR}`)) {
-        copyItemsToSite(dataDir);
-        analysis.updateData(dataDir, (_newItems) => {
+        if (fs.existsSync(`${dataDir}/latest-canonical.json.${analysis.FILE_COMPRESSOR}`)) {
+            copyItemsToSite(dataDir);
+            analysis.updateData(dataDir, (_newItems) => {
+                copyItemsToSite(dataDir);
+            });
+        } else {
+            await analysis.updateData(dataDir);
+            copyItemsToSite(dataDir);
+        }
+        scheduleFunction(5, 0, 0, async () => {
+            items = await analysis.updateData(dataDir);
             copyItemsToSite(dataDir);
         });
-    } else {
-        await analysis.updateData(dataDir);
-        copyItemsToSite(dataDir);
     }
-    scheduleFunction(5, 0, 0, async () => {
-        items = await analysis.updateData(dataDir);
-        copyItemsToSite(dataDir);
-    });
 
     const app = express();
     app.use(compression());
+    app.use(function (req, res, next) {
+        if (req.method == "GET") {
+            if (req.path == "/") {
+                req.url = "/index.html";
+            }
+            if (req.path.endsWith(".html")) {
+                // Only html files are translated
+                let pickedLanguage = req.acceptsLanguages(i18n.locales);
+                if (pickedLanguage) {
+                    let translatedPath = req.path.substring(0, req.path.length - "html".length) + pickedLanguage + ".html";
+                    req.url = translatedPath;
+                } // otherwise use default, untranslated file
+            }
+        }
+        next();
+    });
```

Comment:
Static html files are generated in 2 flavors:
1. Same as before, with german localization, this is the fallback
2. Once again with language tag (`index.cs.html`)

This code intercepts requests to html files and internally rewrites them to point at the localized versions, if there is a supported language in request headers.

This means that the hosting is no longer "dumb". If that is something that we really want, we can do this redirect in JavaScript.